extensions/HuggingFace/typescript/hf.ts

```diff
@@ -148,20 +205,33 @@ export class HuggingFaceTextGenerationModelParser extends aiconfig.Parameterized
     // if no options are passed in, don't stream because streaming is dependent on a callback handler
     const stream = options ? (options.stream ? options.stream : true) : false;
 
+    let output: Output | undefined;
+
     if (stream) {
       const response = await this.hfClient.textGenerationStream(
         textGenerationArgs
       );
-      const output = await ConstructStreamOutput(
+      output = await ConstructStreamOutput(
         response,
         options as InferenceOptions
       );
-      return output;
     } else {
       const response = await this.hfClient.textGeneration(textGenerationArgs);
-      const output = constructOutput(response);
-      return output;
+      output = constructOutput(response);
     }
+
+    prompt.outputs = [output];
```

Comment:
Why do we do this here and not before?

Reply:
Most of these changes are from [PR#283](https://github.com/lastmile-ai/aiconfig/pull/283)